nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Utilities/MSBuildHelper.cs

```diff
@@ -154,7 +154,20 @@ public static IEnumerable<Dependency> GetTopLevelPackageDependenyInfos(Immutable
                 {
                     if (!string.IsNullOrWhiteSpace(attributeValue))
                     {
-                        packageInfo[attributeValue] = versionSpecification;
+                        if (packageInfo.TryGetValue(attributeValue, out var existingInfo))
+                        {
+                            var existingVersion = existingInfo.Item1;
+                            var existingUpdate = existingInfo.Item2;
+                            var vSpec = string.IsNullOrEmpty(versionSpecification) ? existingVersion : versionSpecification;
```

Comment:
If the package reference is an update, should its version be the one that ends up in the packageInfo?